### PR TITLE
Improve PDFPlumber parsing

### LINKED ISSUE #38: `pdfplumber` not detecting table anymore
Also, add a test to check if this happens again.

### PR DESCRIPTION
**Pull request by dilithjay**
- Fix #38 
- Improve table parsing by specifying `table_settings`
- Enable `bold` and `italic` formatting

### CODE REVIEW

File: tests/test_parser.py [hunk 1]
```diff
@@ -177,3 +177,11 @@ async def test_dynamic_js_parsing():
     # Check if the content contains the expected information
     should_contain_info = "6 Types of experimentation"
     assert should_contain_info.lower() in results.strip().lower()
+
+
+@pytest.mark.asyncio
+async def test_table_parsing():
+    sample = "examples/inputs/test_1.pdf"
+    parser_type = "AUTO"
```
Comment (pramitchoudhary) [inline tests/test_parser.py:185]: Shouldn't the test be for `STATIC_PARSE` or both?

File: lexoid/core/parse_type/static_parser.py [hunk 2]
```diff
@@ -89,15 +89,21 @@ def process_table(table) -> str:
 
     # Convert to DataFrame and handle empty cells
     df = pd.DataFrame(table_data)
+    df.replace("", pd.NA, inplace=True)
```
Comment (pramitchoudhary) [inline lexoid/core/parse_type/static_parser.py:92]: Should we also bump the version for future dev/release?

File: lexoid/core/parse_type/static_parser.py [hunk 3]
```diff
@@ -233,46 +286,42 @@ def detect_heading_level(font_size):
                 # If we were collecting a heading
                 if current_heading:
                     level = detect_heading_level(current_heading[0]["size"])
-                    heading_text = " ".join(word["text"] for word in current_heading)
-                    markdown_content.append(f"{'#' * level} {heading_text}\n\n")
+                    heading_text = format_paragraph(current_heading)
```
Comment (pramitchoudhary) [inline lexoid/core/parse_type/static_parser.py:289]: Before after for `test_explicit_hyperlink_n_img.pdf`

Before:
Screenshot 2024-12-27 at 11 05 36 AM

After:
Screenshot 2024-12-27 at 10 49 44 AM

**Review by pramitchoudhary — APPROVED** [review]
thanks @dilithjay 🙏

File: tests/test_parser.py (hunk 1 above)
Comment (dilithjay) [reply tests/test_parser.py:185]: Ah yes, it should be `STATIC_PARSE`. Will fix.

File: lexoid/core/parse_type/static_parser.py (hunk 2 above)
Comment (dilithjay) [reply lexoid/core/parse_type/static_parser.py:92]: You mean the Lexoid version, yes?

File: tests/test_parser.py (hunk 1 above)
Comment (dilithjay) [reply tests/test_parser.py]: Done

File: lexoid/core/parse_type/static_parser.py (hunk 2 above)
Comment (pramitchoudhary) [reply lexoid/core/parse_type/static_parser.py:92]: ya.

**Pull request merged**